src/indexnotation/tensormacros.jl

```diff
@@ -85,6 +86,12 @@ function tensorparser(tensorexpr, kwargs...)
             val in (:warn, :cache) ||
                 throw(ArgumentError("Invalid use of `costcheck`, should be `costcheck=warn` or `costcheck=cache`"))
             parser.contractioncostcheck = val
+        elseif name == :opt_algorithm
```

Comment:
I think here you will have to be a little careful, in principle there is no order to the keyword arguments.
If I am not mistaken, now if the user first supplies `opt=(a = 2, b = 2, ...)`, and only afterwards `opt_algorithm=...`, the algorithm will be ignored.

My best guess is that you probably want to attempt to extract an optimizer and optdict, and only after all kwargs have been parsed, you can construct the `contractiontreebuilder`

Reply:
Thank you very much for pointing that out, I did not notice that perviously.
In the revised version, the `contractiontreebuilder` will be constructed after all other kwargs have been parsed.